lms/djangoapps/grades/new/course_data.py

```diff
@@ -20,23 +25,17 @@ def __init__(self, user, course=None, collected_block_structure=None, course_str
     @property
     def course_key(self):
         if not self._course_key:
-            if self._course:
-                self._course_key = self._course.id
-            else:
-                structure = self._structure or self._collected_block_structure
-                self._course_key = structure.root_block_usage_key.course_key
+            self._course_key = self.course.id
```

Comment:
Ah yes, I remember you'd mentioned that before. My concerns here arose in testing - it seems like it's possible to get differing block structures for 2 otherwise identical `CourseData` objects, based on the order in which the various backing `_something` variables are initialized. @sanfordstudent and I are working through it, I think we'll be able to revert most of this.

Sandy - see https://gist.github.com/efischer19/3f573eddb9fbef4f032eeb2488b2005a